flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/schema/PulsarSchemaUtils.java

```diff
@@ -0,0 +1,183 @@
+        registerSchemaFactory(IntSchemaFactory.class);
+        registerSchemaFactory(LongSchemaFactory.class);
+        registerSchemaFactory(StringSchemaFactory.class);
+        registerSchemaFactory(FloatSchemaFactory.class);
+        registerSchemaFactory(DoubleSchemaFactory.class);
+        registerSchemaFactory(BooleanSchemaFactory.class);
+        registerSchemaFactory(BytesSchemaFactory.class);
+        registerSchemaFactory(NoneSchemaFactory.class);
+        registerSchemaFactory(DateSchemaFactory.class);
+        registerSchemaFactory(TimeSchemaFactory.class);
+        registerSchemaFactory(TimestampSchemaFactory.class);
+        registerSchemaFactory(InstantSchemaFactory.class);
+        registerSchemaFactory(LocalDateSchemaFactory.class);
+        registerSchemaFactory(LocalTimeSchemaFactory.class);
+        registerSchemaFactory(LocalDateTimeSchemaFactory.class);
+        registerSchemaFactory(KeyValueSchemaFactory.class);
+        registerSchemaFactory(ByteBufferSchemaFactory.class);
+        registerSchemaFactory(ByteBufSchemaFactory.class);
+    }
+
+    private PulsarSchemaUtils() {
+        // No need to create instance.
+    }
+
+    private static void registerSchemaFactory(Class<? extends PulsarSchemaFactory> factoryClass) {
+        PulsarSchemaFactory<?> factory = InstantiationUtil.instantiate(factoryClass);
+        Set<PulsarSchemaFactory<?>> factories =
+                FACTORY_REGISTER.computeIfAbsent(factory.type(), f -> new HashSet<>());
+
+        factories.add(factory);
+    }
+
+    /** Extract the defined class type from pulsar schema type. */
+    public static <T> Class<T> extractSchemaClass(Schema<T> schema) {
+        Type argument =
+                ((ParameterizedType) schema.getClass().getGenericSuperclass())
+                        .getActualTypeArguments()[0];
+
+        return TypeExtractionUtils.typeToClass(argument);
+    }
+
+    /**
+     * Pulsar has a hugh set of built-in schemas. We can create them by the given {@link
+     * SchemaInfo}.
+     *
+     * <p>Supported schema check list.
+     *
+     * <pre>{@code
+     * - [ ] AvroSchema
+     * - [x] BooleanSchema
+     * - [x] ByteBufferSchema
+     * - [x] ByteBufSchema
+     * - [x] ByteSchema
+     * - [x] BytesSchema
+     * - [x] DateSchema
+     * - [x] DoubleSchema
+     * - [x] FloatSchema
+     * - [x] InstantSchema
+     * - [x] IntSchema
+     * - [x] KeyValueSchemaImpl
+     * - [ ] JSONSchema
+     * - [x] LocalDateSchema
+     * - [x] LocalDateTimeSchema
+     * - [x] LocalTimeSchema
+     * - [x] LongSchema
+     * - [ ] ProtobufNativeSchema
+     * - [ ] ProtobufSchema
+     * - [x] ShortSchema
+     * - [x] StringSchema
+     * - [x] TimeSchema
+     * - [x] TimestampSchema
+     * }</pre>
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Schema<T> createSchema(SchemaInfo info) {
+        return FACTORY_REGISTER.getOrDefault(info.getType(), emptySet()).stream()
```

Comment:
When do we actually have one than more type for a given `SchemaInfo`? I was thinking that we should rather fail if we have conflicts instead of picking _some_ candidate.